modules/bloxone_infra_host_azure/main.tf

```diff
@@ -0,0 +1,146 @@
+ * A `tf_module_host_id` tag will be added to the tags variable so that the data source can uniquely find the host.
+ *
+ * This module will also create a BloxOne Infra Service for each service type provided in the `services` variable.
+ * The service will be named `<service_type>_<host_display_name>`.
+ *
+ * ## Example Usage
+ *
+ * ```hcl
+ * module "bloxone_infra_host_azure" {
+ *   source = "github.com/infobloxopen/terraform-provider-bloxone//modules/bloxone_infra_host_azure"
+ *
+ *   vm_name                   = "bloxone-vm"
+ *   location                  = "eastus"
+ *   resource_group_name       = "my-resource-group"
+ *   subnet_id                 = "subnet-id"
+ *   vnet_id                   = "vnet-id"
+ *   vm_network_security_group = "nsg-id"
+ * 
+ *   source_image_reference_offer = "infoblox-bloxone-34"
+ *   source_image_reference_version = "3.8.1"
+ *
+ *   azure_instance_tags       = {
+ *     environment = "dev"
+ *   }
+ *
+ *   tags                      = {
+ *     location = "office1"
+ *   }
+ *
+ *   services = {
+ *     dhcp = "start"
+ *     dns = "start"
+ *   }
+ * }
+ * ```
+ * 
+ */
+
+resource "random_uuid" "this" {}
+
+locals {
+  join_token = var.join_token == null ? bloxone_infra_join_token.this[0].join_token : var.join_token
+  tags       = merge(
+    var.tags,
+    {
+      "tf_module_host_id" = "bloxone-host-${random_uuid.this.result}"
+    }
+  )
+}
+
+resource "bloxone_infra_join_token" "this" {
+  count = var.join_token == null ? 1 : 0
+  name  = "jt-${random_uuid.this.result}"
+}
+
+resource "azurerm_network_interface" "this" {
+  name                = "${var.vm_name}-nic"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  tags                = var.azure_instance_tags
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.vnet_subnet_id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "this" {
+  name                            = var.vm_name
+  location                        = var.location
+  resource_group_name             = var.resource_group_name
+  network_interface_ids           = [azurerm_network_interface.this.id]
+  admin_username                  = "dummyuser"
+  size                            = var.vm_size
+  tags                            = var.azure_instance_tags
+
+  admin_ssh_key {
+    username   = "dummyuser"
+    public_key = file(var.ssh_public_key_path)
+  }
+  
+  os_disk {
+    name                 = "${var.vm_name}-os-disk"
+    caching              = "ReadWrite"
+    storage_account_type = var.os_disk_storage_account_type
+  }
+
+  source_image_reference {
+    publisher = var.source_image_reference_publisher
+    offer     = var.source_image_reference_offer
+    sku       = var.source_image_reference_sku
+    version   = var.source_image_reference_version
+  }
+
+  plan {
```

Comment:
why do you need both `plan` and `source_image_reference`. To me it looks like they have the same information?